src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEnginePublisher.java

```diff
@@ -315,15 +338,57 @@ public FormValidation doCheckZone(
       return FormValidation.ok();
     }
 
-    public FormValidation doCheckProjectId(@QueryParameter String value) {
-      if (Strings.isNullOrEmpty(value)) {
+    public ListBoxModel doFillProjectIdItems(
+        @AncestorInPath Jenkins context,
+        @QueryParameter("credentialsId") final String credentialsId) {
+      ListBoxModel items = new ListBoxModel();
+      items.add("");
+      if (Strings.isNullOrEmpty(credentialsId)) {
+        return items;
+      }
+
+      ClientFactory clientFactory;
+      try {
+        clientFactory = this.getClientFactory(context, credentialsId);
+      } catch (AbortException ae) {
+        items.clear();
+        items.add(ae.getMessage(), "");
+        return items;
+      }
+
+      String defaultProjectId = clientFactory.getDefaultProjectId();
+      try {
+        CloudResourceManagerClient client = clientFactory.cloudResourceManagerClient();
+        List<Project> projects = client.getAccountProjects();
+
+        if (!projects.isEmpty()) {
+          projects
+              .stream()
+              .filter(p -> !p.getProjectId().equals(defaultProjectId))
+              .forEach(p -> items.add(p.getProjectId()));
+          if (projects.size() > items.size() - 1 && !Strings.isNullOrEmpty(defaultProjectId)) {
+            items.add(new Option(defaultProjectId, defaultProjectId, true));
+          } else {
+            // Select not the first (empty) item but the second item, which exists because
+            // projects is not empty.
+            items.get(1).selected = true;
+          }
+        }
+        return items;
+      } catch (IOException ioe) {
```

Comment:
Swallowing the exception here is a red flag. This should be logged.

Reply:
Added the logging. In addition, during verification there will be a similar error message that basically indicates "Could not retrieve project IDs for verification."

Reply:
Thanks for adding the log. If you use this log override, you can pass the Throwable as an arg and let the logging framework handle the formatting. Also, please make sure to include a descriptive log message for the user.

See: https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#log(java.util.logging.Level,%20java.lang.String,%20java.lang.Throwable)